src/components/bcros/filing/common/HeaderActions.vue

```diff
@@ -369,5 +379,11 @@ const handleButtonClick = async () => {
 
     ui.fetchingData = false
   }
+
+  // If document records are enabled and the documents list is empty,
+  // fetch the document records for the current business identifier.
+  if (enableDocumentRecords.value && !documents.value?.length) {
+    await getCorpDocuments({ consumerIdentifier: currentBusinessIdentifier.value })
+  }
```

Comment:
I think this should be moved to the block above, or at least within the spinner's scope. This will keep the spinner visible while documents are being loaded.

Reply:
This would tie it to the filings documents (outputs), ie no outputs, no request to DRS. I don't know it well enough to say thats always the case. 
So we probably want to increase the scope of the spinner?

Reply:
Looks good.